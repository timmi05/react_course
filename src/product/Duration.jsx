import React from 'react';
import {DigitalInfo} from '../component/DigitalInfo';

export const Duration = props => {
    const {runtime} = props;
    if (runtime === 0) {
        return (
            <a/>
        )
    }
    return (
        <DigitalInfo
            value={runtime}
            meaning=' min'
        />
    )
};