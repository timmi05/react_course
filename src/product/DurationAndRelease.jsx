import React from 'react';
import DigitalInfo from '../component/DigitalInfo';

const DurationAndRelease = (props) =>{
    const { runtime, release_date } = props;
    return (
         <div className="duration-and-release">
            <DigitalInfo
            value = { runtime }
            meaning = 'year'
            />
            <DigitalInfo
            value = {runtime}
            meaning = 'min'
            />
         </div>
    );
}

export default DurationAndRelease