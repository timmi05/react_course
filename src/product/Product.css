.product{
    display: flex;
    flex-direction: row;
    margin-bottom: 24px;
    margin-top: 24px;
}

.product-cover{
    height: 450px;
    width: 315px;
}