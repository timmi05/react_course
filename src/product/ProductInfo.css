.product-info {
    padding: 30px 50px;
    width: 80%;
}

.product-info-title{
    font-size: 34px;
    padding-right: 20px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}

.product-info-title-rating{
    display: flex;
    flex-direction: row;
}

.product-info-tagline{
    margin-bottom: 25px;
    font-size: 20px;
}