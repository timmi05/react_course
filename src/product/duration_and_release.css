.duration-and-release{
    color: #f65261;
    font-size: 26px;
    margin-bottom: 25px;
}

.duration-and-release-release{
    margin-right: 30px;
}