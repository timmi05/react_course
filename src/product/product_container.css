.product-container{
    display: flex;
    flex-direction: row;
}

.product-container-cover{
    height: 400px;
    width: 250px;
}