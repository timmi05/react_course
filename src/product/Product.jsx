import React from 'react';
import {ProductInfo} from './ProductInfo';
import './Product.css';

export const Product = props =>
    <div className='product'>
        <img className='product-cover' src={props.movie.poster_path} alt={props.movie.title}/>
        <ProductInfo
            {...props}
        />
    </div>;
