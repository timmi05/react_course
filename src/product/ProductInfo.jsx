import React from 'react';
import {ProductRating} from './ProductRating';
import {ProductDetails} from './ProductDetails';
import {ProductDescription} from './ProductDescription';
import './ProductInfo.css'

export const ProductInfo = props => {
    const {movie} = props;
    return (
        <div className='product-info'>
            <div className='product-info-title-rating'>
                <span className='product-info-title'>{movie.title}</span>
                <ProductRating
                    rating={movie.vote_average}
                />
            </div>
            <div className='product-info-tagline'>{movie.tagline}</div>
            <ProductDetails
                runtime={movie.runtime}
                releaseDate={movie.release_date.substring(0, 4)}
            />
            <ProductDescription
                overview={movie.overview}
            />
        </div>
    );
};
