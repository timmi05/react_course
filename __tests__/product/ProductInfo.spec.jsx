import React from 'react';
import { shallow } from 'enzyme';
import { ProductInfo } from '../../src/product/ProductInfo';

describe('ProductInfo rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<ProductInfo/>);
        expect(component).toMatchSnapshot();
    });
});
