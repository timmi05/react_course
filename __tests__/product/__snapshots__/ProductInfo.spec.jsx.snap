// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo rendering renders correctly 1`] = `
<div
  className="product-info"
>
  <div
    className="product-info-title-rating"
  >
    <span
      className="product-info-title"
    >
      title
    </span>
    <ProductRating />
  </div>
  <div
    className="product-info-tagline"
  >
    tagline
  </div>
  <ProductDetails
    releaseDate="rele"
  />
  <ProductDescription />
</div>
`;
